### PR TITLE
Fix center afpoints

### PR DESCRIPTION
**Pull request by rderimay**
Modified DefaultPointRenderer to assume that the focusPoints are now
the center of the points and not the left top corner anymore.
Temporarily patched the DefaultDelegate to artificially add back half
of the focus point dimensions until the config files have been updated.
Designed a new focus_point image to better accommodate the fact that
the picture can not be resized to the right size of the focus point.

### CODE REVIEW

File: focuspoints.lrdevplugin/DefaultDelegates.lua [hunk 1]
```diff
@@ -43,9 +44,11 @@ function DefaultDelegates.getDefaultAfPoints(photo, metaData)
     LrErrors.throwUserError("Unable to find any AF point info within the file.")
     return nil, nil
   end
-  
-  local x = DefaultDelegates.focusPointsMap[focusPoint][1]
-  local y = DefaultDelegates.focusPointsMap[focusPoint][2]
+
+  -- TODO: The addition of the dimension should be removed once all config files have been
+  -- updated to reflect the center of the focus points
+  local x = DefaultDelegates.focusPointsMap[focusPoint][1] + DefaultDelegates.focusPointDimen[1]
```
Comment (musselwhizzle) [inline focuspoints.lrdevplugin/DefaultDelegates.lua]: this should be 1/2 the size of the focus point's width to get the center.

File: focuspoints.lrdevplugin/DefaultDelegates.lua [hunk 2]
```diff
@@ -24,6 +24,7 @@ require "Utils"
 
 DefaultDelegates = {}
 DefaultDelegates.focusPointsMap = nil
+DefaultDelegates.focusPointDimen = nil
```
Comment (musselwhizzle) [inline focuspoints.lrdevplugin/DefaultDelegates.lua]: I'm not seeing where this is ever set. Can you show/tell me?

Comment (musselwhizzle) [inline focuspoints.lrdevplugin/DefaultDelegates.lua]: I merged this into a branch "center_it" to run it. the property is null at run time. I can take care of it sometime or feel free or you get time.

**Comment by musselwhizzle** [comment]
using the code, my focus points are off by a small margin as where before they were dead on. Perhaps my focusPointDimens are the wrong size. Before I merge this over into master I want to get the points dead on accurate again. Will merge once I have time to get the camera and reshoot some photos on rulers to measure.

File: focuspoints.lrdevplugin/DefaultDelegates.lua (hunk 1 above)
Comment (rderimay) [inline focuspoints.lrdevplugin/DefaultDelegates.lua]: Correct !

File: focuspoints.lrdevplugin/DefaultDelegates.lua (hunk 2 above)
Comment (rderimay) [inline focuspoints.lrdevplugin/DefaultDelegates.lua]: yeah. I saw you added it back in the PointsRedererFactory. No idea why it is not on GitHub but in my code...

**Comment by rderimay** [comment]
Ok. starting to work on the rotation algorithm. I'll if I can reach something better...
For the AF-Points images, I dont think this is a good idea to have replaced my X and Y offsets by the X and Y sizes of the pictures. In the future, we may want to have af-point pictures where the anchor point is not in the center.
If you are ok, I will add a parameter to the delegate to have the possibility to have different af-points pictures for different cameras, defaulting to yours.

**Comment by rderimay** [comment]
ok @musselwhizzle, rotation problem solved. Working like a charm.
I will make a bit of refactoring before integrating.
Do you have picture with an exif rotation inside to test that the algorithm is still working when (shotOrientation == 90 or (shotOrientation == 270) ?

**Comment by musselwhizzle** [comment]
all good suggestions and I'll take them all. Let me add 2 vertical photos to the image repo I made for us. 4718 and 4720 in the repo are 90 degrees and 270 degrees images. I hope to reshoot my camera today and validate the new adjustment on the center points. Once I do that, we can merge this into master.

**Comment by rderimay** [comment]
I did not find any way to know if the user has rotated the pictures in the grid view using the small left and right arrows. Do you have an idea ?

**Comment by musselwhizzle** [comment]
Correct. Lightroom doesn't report this info. Not much you can do here. Its
marked as a known issue in the readme.

**Pull request merged**